### PR TITLE
Move origin of chef CQC to skillchip

### PR DESCRIPTION
**Pull request by coiax**
:cl:
tweak: The chef's CQC (close quarters cooking) ability is now imparted
by a skillchip in their brain, similar to Station Engineer's skillchip
for wire colours.
/:cl:

One more intrinsic property of a job that has been moved to an
extrinsic, that won't likely affect many rounds, but now makes more
aspects "lootable".

### CODE REVIEW

File: code/modules/library/skill_learning/job_skillchips/chef.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	skill_name = "Close Quarters Cooking"
+	skill_description = "A specialised form of self defence, developed by skilled sous-chef de cuisines. No man fights harder than a chef to defend his kitchen."
+	skill_icon = "utensils"
+	activate_message = "<span class='notice'>You can visualize how to defend your kitchen with martial arts.</span>"
+	deactivate_message = "<span class='notice'>You forget how to control your muscles to execute kicks, slams and restraints while in a kitchen environment.</span>"
+	var/datum/martial_art/cqc/under_siege/style
+
+/obj/item/skillchip/job/chef/Initialize()
+	. = ..()
+	style = new
```
Comment (coiax) [inline code/modules/library/skill_learning/job_skillchips/chef.dm:13]: The format of how this skillchip works technically, is copied from how "krav maga gloves" currently work. If there's a better way to phrase it, or if there's some issues with Destroy, those gloves will need to be updated too.

**Comment by Pugie** [comment]
lmfao

File: code/modules/library/skill_learning/job_skillchips/chef.dm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/obj/item/skillchip/job/chef
+	name = "B0RK-X3 skillchip"  // bork bork bork
+	desc = "This biochip faintly smells of garlic, which is odd for something that is normally wedged inside a user's brain. Consult a dietician before use."
+	skill_name = "Close Quarters Cooking"
+	skill_description = "A specialised form of self defence, developed by skilled sous-chef de cuisines. No man fights harder than a chef to defend his kitchen."
+	skill_icon = "utensils"
+	activate_message = "<span class='notice'>You can visualize how to defend your kitchen with martial arts.</span>"
```
Comment (Iamgoofball) [inline code/modules/library/skill_learning/job_skillchips/chef.dm:7]:
```suggestion
	activate_message = "<span class='notice'>You can now operate under siege.</span>"
```

File: code/modules/library/skill_learning/job_skillchips/chef.dm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	deactivate_message = "<span class='notice'>You forget how to control your muscles to execute kicks, slams and restraints while in a kitchen environment.</span>"
```
Comment (Iamgoofball) [inline code/modules/library/skill_learning/job_skillchips/chef.dm:8]:
```suggestion
	deactivate_message = "<span class='notice'>You're no longer capable of operating while under siege.</span>"
```

**Comment by JoshAdamPowell** [comment]
Should the HoP have a spare like CE/RD does with their department chips? Not like it's super useful for anyone other than a chef but if the HoP wants to promote more...

**Comment by Cyberboss** [comment]
I think the `under_siege` martial art was only used here, you can remove it with this change.

**Comment by Iamgoofball** [comment]
Uh, no, no you cannot. It has the kitchen limitation and some balance changes.

File: code/modules/library/skill_learning/job_skillchips/chef.dm [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	. = ..()
+	style = new
+
+/obj/item/skillchip/job/chef/on_activate(mob/living/carbon/user, silent = FALSE)
+	. = ..()
+	style.teach(user, make_temporary = TRUE)
+
+/obj/item/skillchip/job/chef/on_deactivate(mob/living/carbon/user, silent = FALSE)
+	style.remove(user)
+	. = ..()
```
Comment (Rohesie) [inline code/modules/library/skill_learning/job_skillchips/chef.dm:21]:
```suggestion
	return ..()
```

**Comment by jacklecroy** [comment]
next up languages for the curator

**Comment by jacklecroy** [comment]
> 
> 
> I think the `under_siege` martial art was only used here, you can remove it with this change.

The skillchip is still using it

**Comment by Pugie** [comment]
skilled professionals dont exist in the future that's failrp and unforgiveable everything is controlled by rp nanomachines

**Comment by Investigator77** [comment]
> skilled professionals dont exist in the future that's failrp and unforgiveable everything is controlled by rp nanomachines

Based reality

**Pull request merged**